File: boj/2023/solved.ac/2-230619_0627/_18111.cpp
#include <cstdio>

int N, M, B;
int H[500][500];

int f(int h)
{
	int lo = 0, hi = 0;
	for (int y = 0; y < N; y++) {
		for (int x = 0; x < M; x++) {
			if (H[y][x] <= h)
				lo += h - H[y][x];
			else
				hi += H[y][x] - h;
		}
	}
	return B + hi - lo >= 0 ? lo + 2 * hi : -1;
}
int main()
{
	scanf("%d %d %d", &N, &M, &B);
	for (int y = 0; y < N; y++) {
		for (int x = 0; x < M; x++)
			scanf("%d", &H[y][x]);
	}

	unsigned ans_t = -1, ans_h, t;
	for (int h = 256; h >= 0; h--) {
		if ((t = f(h)) < ans_t)
			ans_t = t, ans_h = h;
	}
	printf("%d %d", ans_t, ans_h);
}

File: boj/solved.ac/2-230619_0627/_18111.py
# 파이썬이 느려서 올바르게 짜도 시간 초과가 남. (PyPy 말고는 답 없음)

import sys
input = lambda: sys.stdin.readline().rstrip()

from math import inf

N, M, B = map(int, input().split())
H = [ list(map(int, input().split())) for _ in range(N) ]

def f(h):
	lo, hi = 0, 0
	for y in range(N):
		for x in range(M):
			if H[y][x] <= h:
				lo += h - H[y][x]
			else:
				hi += H[y][x] - h
	return lo + 2 * hi if B + hi - lo >= 0 else inf

ans_t, ans_h = inf, None
for h in range(256, -1, -1):
	t = f(h)
	if t < ans_t:
		ans_t, ans_h = t, h
print(ans_t, ans_h)